src/document/DocumentAnnotator.ts

```diff
@@ -1,21 +1,46 @@
+import debounce from 'lodash/debounce';
 import BaseAnnotator, { Options } from '../common/BaseAnnotator';
 import BaseManager from '../common/BaseManager';
 import { centerRegion, isRegion, RegionManager } from '../region';
 import { Event } from '../@types';
 import { getAnnotation } from '../store/annotations';
-import { Mode } from '../store';
+import { getSelectionItem } from './docUtil';
+import { Mode, setSelectionAction } from '../store';
 import { scrollToLocation } from '../utils/scroll';
 import './DocumentAnnotator.scss';
 
+const TEXT_LAYER_ENHANCEMENT = 300;
+
 export default class DocumentAnnotator extends BaseAnnotator {
     annotatedEl?: HTMLElement;
 
     managers: Map<number, Set<BaseManager>> = new Map();
 
+    selectionChangeTimer: number | null = null;
+
     constructor(options: Options) {
         super(options);
 
         this.addListener(Event.ANNOTATIONS_MODE_CHANGE, this.handleChangeMode);
+        document.addEventListener('selectionchange', this.handleSelectionChange);
```

Comment:
Should we do something to conditionally add this event listener only if there is a text layer rendered by pdfjs?

Reply:
We cannot do it in constructor because the textlayers haven't been generated yet. We can only do it in `init()`. But that is not ideal because every zoom in/out will trigger `init()` and we will add multiple listeners.

Reply:
I think we can add one more option `hasTextLayer` in constructor when construct annotator in content-preview and use that to conditionally add this event. What do you think?

Reply:
Maybe we can address it when we also address the highlight feature flip being passed into annotations